x-pack/plugins/maps/public/connected_components/toolbar_overlay/feature_draw_controls/feature_edit_tools/feature_edit_tools.tsx

```diff
@@ -36,6 +36,13 @@ export function FeatureEditTools(props: Props) {
   const drawPointSelected = props.drawShape === DRAW_SHAPE.POINT;
   const deleteSelected = props.drawShape === DRAW_SHAPE.DELETE;
 
+  function toggleDrawShape(mode: DRAW_SHAPE) {
```

Comment:
Can you change to the below implementation to avoid calling setDrawShape twice?

```
function toggleDrawShape(mode: DRAW_SHAPE) {
    if (mode && props.drawShape === mode) {
      props.setDrawShape(null);
    } else {
      props.setDrawShape(mode); 
    }
  }
```